src/index.native.scroll.js

```diff
@@ -103,7 +103,9 @@ class SwipeableViews extends React.Component {
       index,
     } = nextProps;
 
-    if (typeof index === 'number' && index !== this.props.index) {
+    // Sometimes the index can be the same, but the slide
+    // may have changed, so just always set it
+    if (typeof index === 'number') {
```

Comment:
Tested with that new addition and all appears to be working! Not entirely sure what was causing the issue previously.